code/modules/mob/living/simple_animal/friendly/mouse.dm

```diff
@@ -1,62 +1,107 @@
+	'sound/effects/creatures/mouse_squeaks_4.ogg')
+	var/last_softsqueak = null//Used to prevent the same soft squeak twice in a row
+	var/squeals = 5//Spam control.
+	var/maxSqueals = 5//SPAM PROTECTION
+	var/last_squealgain = 0// #TODO-FUTURE: Remove from life() once something else is created
+	var/squeakcooldown = 0
 	pass_flags = PASSTABLE
-	speak_chance = 1
+	speak_chance = 5
 	turns_per_move = 5
 	see_in_dark = 6
 	maxHealth = 5
 	health = 5
 	meat_type = /obj/item/weapon/reagent_containers/food/snacks/meat
 	response_help  = "pets"
 	response_disarm = "gently pushes aside"
-	response_harm   = "stamps on"
+	response_harm   = "stomps on"
 	density = 0
+	meat_amount = 1
 	var/body_color //brown, gray and white, leave blank for random
 	layer = MOB_LAYER
+	mob_size = MOB_MINISCULE
 	min_oxy = 16 //Require atleast 16kPA oxygen
 	minbodytemp = 223		//Below -50 Degrees Celcius
 	maxbodytemp = 323	//Above 50 Degrees Celcius
 	universal_speak = 0
 	universal_understand = 1
 	holder_type = /obj/item/weapon/holder/mouse
-	mob_size = MOB_MINISCULE
-	possession_candidate = 1
+	//digest_factor = 0.05 //These are for animal-eating code, not yet ported
+	//min_scan_interval = 2
+	//max_scan_interval = 20
+	//seek_speed = 1
 
-	can_pull_size = ITEM_SIZE_TINY
+	can_pull_size = 1
 	can_pull_mobs = MOB_PULL_NONE
 
+	var/decompose_time = 18000
+
+	//kitchen_tag = "rodent" //This is part of cooking overhaul, not yet ported
+
 /mob/living/simple_animal/mouse/Life()
-	..()
-	if(!stat && prob(speak_chance))
-		for(var/mob/M in view())
-			M << 'sound/effects/mousesqueek.ogg'
-
-	if(!ckey && stat == CONSCIOUS && prob(0.5))
-		stat = UNCONSCIOUS
-		icon_state = "mouse_[body_color]_sleep"
-		wander = 0
-		speak_chance = 0
-		//snuffles
-	else if(stat == UNCONSCIOUS)
-		if(ckey || prob(1))
-			stat = CONSCIOUS
-			icon_state = "mouse_[body_color]"
-			wander = 1
-		else if(prob(5))
-			visible_emote("snuffles.")
-
-/mob/living/simple_animal/mouse/lay_down()
-	..()
-	icon_state = resting ? "mouse_[body_color]_sleep" : "mouse_[body_color]"
+	if(..())
 
-/mob/living/simple_animal/mouse/New()
-	..()
+		if(client)
+			walk_to(src,0)
+
+			//Player-animals don't do random speech normally, so this is here
+			//Player-controlled mice will still squeak, but less often than NPC mice
+			if (stat == CONSCIOUS && prob(speak_chance*0.05))
+				squeak_soft(0)
+
+			//if(is_ventcrawling == 0) //Part of ventcrawling rework, not yet ported
+				//sight = initial(sight) // Returns mouse sight to normal when they leave a vent
+
+			if (squeals < maxSqueals)
+				var/diff = world.time - last_squealgain
+				if (diff > 600)
+					squeals++
+					last_squealgain = world.time
+
+	else
+		if ((world.time - timeofdeath) > decompose_time)
+			dust()
+
+/mob/living/simple_animal/mouse/Destroy()
+
+	return ..()
 
+//Pixel offsetting as they scamper around
+/mob/living/simple_animal/mouse/Move()
+	if(..())
+		if (prob(50))
+			var/new_pixelx = pixel_x
```

Comment:
These pixel offsets cause mice to look more like a swarm in larger numbers as they expand out over tile space 

https://i.imgur.com/ZRixEoo.png

It also prevents them from stacking ontop of each other when they inhabit the same tile, making it easier to see how many there are